normalizer/core.py

```diff
@@ -171,26 +330,61 @@ def normalize(
         + '\n'
     )
 
+    hubble_score_metrics = {
+        'dockerfile_exists': dockerfile_path.exists(),
+        'manifest_exists': manifest_path.exists(),
+        'config_exists': config_path.exists(),
+        'readme_exists': readme_path.exists(),
+        'requirements_exists': requirements_path.exists(),
+        'tests_exists': bool(test_glob),
+    }
+
     # if not requirements_path.exists():
     #     requirements_path.touch()
 
     # manifest = load_manifest(manifest_path)
 
-    executor = None
+    # inspect executor
+    executors = inspect_executors(py_glob)
+    if len(executors) == 0:
+        raise ExecutorNotFoundError
+    if len(executors) > 1:
+        raise ExecutorExistsError
+
+    executors = filter_executors(executors)
+    if len(executors) == 0:
+        raise IllegalExecutorError
+
+    executor, filepath, docstring, init, endpoints = executors[0]
+    if init:
+        init_args, init_args_defaults, init_annotations, init_docstring = init
+        init_args, init_kwargs = _get_args_kwargs(
+            init_args, init_args_defaults, init_annotations
+        )
+        init = (init_args, init_kwargs, init_docstring)
+
+    for i, endpoint in enumerate(endpoints):
+        if endpoint is not None:
+            (
+                endpoint_name,
+                endpoint_args,
+                endpoint_args_defaults,
+                endpoint_annotations,
+                endpoint_docstring,
+                endpoint_requests,
+            ) = endpoint
+            endpoint_args, endpoint_kwargs = _get_args_kwargs(
+                endpoint_args, endpoint_args_defaults, endpoint_annotations
+            )
+            endpoints[i] = (
+                endpoint_name,
+                endpoint_args,
+                endpoint_kwargs,
+                endpoint_docstring,
+                endpoint_requests,
+            )
```

Comment:
There also seems to be no new logic here - could this go in `inspect_executors`?

Reply:
refactored